json_serializable/lib/src/type_helpers/record_helper.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:analyzer/dart/element/type.dart';
+import 'package:source_helper/source_helper.dart';
+
+import '../type_helper.dart';
+import '../utils.dart';
+
+class RecordHelper extends TypeHelper<TypeHelperContextWithConfig> {
+  const RecordHelper();
+
+  @override
+  Object? deserialize(
+    DartType targetType,
+    String expression,
+    TypeHelperContextWithConfig context,
+    bool defaultProvided,
+  ) {
+    if (targetType is! RecordType) return null;
+
+    final items = <Object>[];
+
+    const paramName = r'$jsonValue';
+
+    var index = 1;
+    for (var field in targetType.positionalFields) {
```

Comment:
It feels a bit weird to serialize the positional fields as a map with the `$*` names. Although it does simplify things some.

I could also see serializing them as a List, with positional fields followed by a nested Map for the named fields, so like (1, 2, {name: 'Foo'}) would be like `[1, 2, {"name": "Foo"}]`.

Ultimately I don't know how much it matters how we serialize records, but that would be a bit more true to the original format?

Reply:
I don't think we gain anything by encoding to look like the definition syntax. It makes encode and decode more complex. I'm trying to do the thing that's just easiest.